src/examples/hotgym/HelloSPTP.cpp

```diff
@@ -148,19 +158,22 @@ EPOCHS = 2; // make test faster in Debug
     //Anomaly (pure x likelihood)
     an = tm.anomaly;
     avgAnom10.compute(an); //moving average
-    if(e % 1000 == 0) {
+    if(e % (500 + avgAnom10.getData().size() /*size of avg window for an*/) == 0) {
       NTA_CHECK(avgAnomOld_ >= avgAnom10.getCurrentAvg()) << "TM should learn and avg anomalies improve, but we got: "
         << avgAnomOld_ << " and now: " << avgAnom10.getCurrentAvg(); //invariant
       avgAnomOld_ = avgAnom10.getCurrentAvg(); //update
     }
-    tAnLikelihood.start();
-    anLikelihood.anomalyProbability(an); //FIXME AnLikelihood is 0.0, probably not working correctly
-    tAnLikelihood.stop();
+
+    //Classifier, Predictor
+    tCls.start();
+    pred.learn(e, outTM, { realToCategory_(data) }); //FIXME fails with bad_alloc if label is too large! PDF should use map, instead of a vector
```

Comment:
FIXME: this should be resolved in the Predictor. A large label should not crash the program (caused by huge PDF if label is large). See if PDF can use map.